youtube_dl/extractor/caffeine.py

```diff
@@ -0,0 +1,43 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    int_or_none,
+)
+
+import re
+
+
+class CaffeineIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?caffeine\.tv/.+/video/(?P<video_id>[0-9a-f-]+)'
+    _TEST = {
+        'url': 'https://www.caffeine.tv/TsuSurf/video/cffc0a00-e73f-11ec-8080-80017d29f26e',
+        'info_dict': {
+            'id': 'cffc0a00-e73f-11ec-8080-80017d29f26e',
+            'ext': 'mp4',
+            'title': 'GOOOOD MORNINNNNN #highlights',
+            'uploader': 'TsuSurf',
+            'duration': 3145,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = re.match(self._VALID_URL, url).group('video_id')
+        json_data = self._download_json('https://api.caffeine.tv/social/public/activity/' + video_id, video_id)
+        broadcast_info = json_data.get('broadcast_info')
+        title = broadcast_info['broadcast_title']
+        video_url = broadcast_info['video_url']
```

Comment:
In case this API call goes wrong, let's make it fail in the simplest way (`traverse_obj()`, `T()` imported from `..utils`):
```suggestion
        broadcast_info = traverse_obj(json_data, ('broadcast_info', T(dict))) or {}
        title = broadcast_info['broadcast_title']
        video_url = broadcast_info['video_url']
```